internal/session/connection_repository.go

```diff
@@ -0,0 +1,69 @@
+package session
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+)
+
+// Repository is the session database repository
+type ConnectionRepository struct {
+	reader db.Reader
+	writer db.Writer
+	kms    *kms.Kms
+
+	// defaultLimit provides a default for limiting the number of results returned from the repo
+	defaultLimit int
+}
+
+// NewRepository creates a new session Connection Repository. Supports the options: WithLimit
+// which sets a default limit on results returned by repo operations.
+func NewConnectionRepository(r db.Reader, w db.Writer, kms *kms.Kms, opt ...Option) (*ConnectionRepository, error) {
+	const op = "sessionConnection.NewRepository"
+	if r == nil {
+		return nil, errors.NewDeprecated(errors.InvalidParameter, op, "nil reader")
```

Comment:
it would be nice to not create new errors using `NewDeprecated`, do to so I think requires that a `context.Context` is passed in here, which would rippled back up into the repo factory functions in other packages, but I think that should be do-able as part of this.